lib/traversing.js

```diff
@@ -9,30 +9,34 @@
 function traverse(node, opt_onEnter, opt_onLeave) {
   if (opt_onEnter) opt_onEnter(node);
 
-  const childNodes = _collectChildNodes(node);
-  childNodes.forEach(function(childNode) {
-    traverse(childNode, opt_onEnter, opt_onLeave);
+  const childNodeInfo = _collectChildNodeInfo(node);
+  childNodeInfo.forEach(function([childNode, propName, parentType]) {
```

Comment:
```suggestion
  childNodeInfo.forEach(([childNode, parentPropName = null, parentType = null]) {
```

Reply:
Re: the test cases, using properties of `NodeType`, good point--I've included that in my latest force push.

Given default parameters in ES6 only working with `undefined`, the usefulness of `null` has, imo, unfortunately been minimized both for semantic clarity (i.e., meaning a deliberately nullish value) and for utility (APIs can't allow or take advantage of `null` for defaults so easily). But in this case, I don't see a need for the handlers to default to anything else, so I'm fine with switching to `null`.

I've added this as a separate commit in case there were any objections, but it's fine by me (FWIW, the defaulting to `null` occurs elsewhere in source, but I got your idea).